src/Illuminate/Notifications/HasDatabaseNotifications.php

```diff
@@ -11,7 +11,9 @@ trait HasDatabaseNotifications
      */
     public function notifications()
     {
-        return $this->morphMany(DatabaseNotification::class, 'notifiable')->latest();
+        return $this->setConnection($this->getNotifiableConnectionName())
```

Comment:
After reviewing this, I'm not sure this is the right approach as this will set the connection on the model class that has `HasDatabaseNotifications` applied. What you want to do instead is set the connection on the retrieved `DatabaseNotification` instances. But that can't be done here.